test/gurka/test_top_speed.cc

```diff
@@ -0,0 +1,76 @@
+#include "gurka.h"
+#include <boost/format.hpp>
+#include <gtest/gtest.h>
+
+using namespace valhalla;
+
+class TopSpeedTest : public ::testing::Test {
+protected:
+  static gurka::map speed_map;
+
+  static void SetUpTestSuite() {
+    constexpr double gridsize = 10;
+
+    const std::string ascii_map = R"(
+      B------C
+      |      |
+      A      D
+      |      |
+      E------F
+    )";
+
+    const gurka::ways ways = {{"ABCD", {{"highway", "residential"}, {"maxspeed", "45"}}},
+                              {"AEFD", {{"highway", "residential"}, {"maxspeed", "20"}}}};
+
+    const auto layout = gurka::detail::map_to_coordinates(ascii_map, gridsize);
+    speed_map = gurka::buildtiles(layout, ways, {}, {}, "test/data/topspeed");
+  }
+
+  inline float getDuration(const valhalla::Api& route) {
+    return route.directions().routes(0).legs(0).summary().time();
+  }
+
+  void doTests(const std::string& costing,
+               const std::unordered_map<std::string, std::string>& options) {
+
+    valhalla::Api default_route = gurka::route(speed_map, "A", "D", costing);
+    float default_time = getDuration(default_route);
+
+    valhalla::Api capped_route = gurka::route(speed_map, "A", "D", costing, options);
+    float capped_time = getDuration(capped_route);
+
+    gurka::assert::raw::expect_eta(capped_route, 19.8f, 0.001f);
+    gurka::assert::raw::expect_path(capped_route, {"AEFD"});
+    ASSERT_GT(capped_time, default_time);
+  }
+};
+
+gurka::map TopSpeedTest::speed_map = {};
+
+TEST_F(TopSpeedTest, AutoTopSpeed) {
+  doTests("auto", {{"/costing_options/auto/top_speed", "20"}});
+}
+
+TEST_F(TopSpeedTest, TruckTopSpeed) {
+  doTests("truck", {{"/costing_options/truck/top_speed", "20"}});
+}
+
+TEST_F(TopSpeedTest, ScooterTopSpeed) {
```

Comment:
Would like to test that `kDefaultTopSpeed` is taken when `top_speed` is outside the acceptable range.. how do I access `top_speed_` (or `options().top_speed()`) from a `valhalla::Api` object?

Reply:
Ehh, was covered by the failing inline test.. not sure I actually fixed the test..